kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/ClusterResolutionResult.java

```diff
@@ -6,109 +6,197 @@
 
 package io.kroxylicious.kubernetes.operator.resolver;
 
+import java.util.List;
 import java.util.Objects;
-import java.util.Set;
-import java.util.stream.Collectors;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
 import java.util.stream.Stream;
 
 import io.kroxylicious.kubernetes.api.common.LocalRef;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaService;
 import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.KafkaProtocolFilter;
+import io.kroxylicious.kubernetes.operator.ResourcesUtil;
+import io.kroxylicious.kubernetes.operator.model.ResolvedCluster;
 
 /**
  * The result of a deep resolution of the dependencies of a single VirtualKafkaCluster. Describes
  * whether there were any resolution problems like dangling references, or referenced resources
  * that have a ResolvedRefs: False condition.
+ *
+ * @param cluster the cluster resource being recursively resolved
+ * @param proxyResolutionResult the resolution result for the KafkaProxy resource this cluster references
+ * @param filterResolutionResults resolution results for all filters referenced by the cluster, in the same order they are declared in the VirtualKafkaCluster spec
+ * @param serviceResolutionResult the resolution result for service referenced by the cluster
+ * @param ingressResolutionResults the resolution results for ingresses referenced by the cluster, in the same order they are declared in the VirtualKafkaCluster spec
  */
 public record ClusterResolutionResult(VirtualKafkaCluster cluster,
-                                      Set<DanglingReference> danglingReferences,
-                                      Set<LocalRef<?>> resourcesWithResolvedRefsFalse,
-                                      Set<LocalRef<?>> referentsWithStaleStatus) {
-    public ClusterResolutionResult {
-        Objects.requireNonNull(danglingReferences);
+                                      ResolutionResult<KafkaProxy> proxyResolutionResult,
+                                      List<ResolutionResult<KafkaProtocolFilter>> filterResolutionResults,
+                                      ResolutionResult<KafkaService> serviceResolutionResult,
+                                      List<IngressResolutionResult> ingressResolutionResults) {
+
+    /**
+     * All referents and cluster are fully resolved iff
+     * <ol>
+     *     <li>All references can be retrieved from kubernetes (we have no dangling refs)</li>
+     *     <li>No referent has a ResolvedRefs=False condition, which declares that it has unresolved dependencies</li>
```

Comment:
And the condition is up to date?

Reply:
good point, I don't think our code considers that currently.

Reply:
created #2146